paasta_tools/kubernetes_tools.py

```diff
@@ -670,7 +692,13 @@ def get_desired_instances(self) -> int:
         """ For now if we have an EBS instance it means we can only have 1 instance
         since we can't attach to multiple instances. In the future we might support
         statefulsets which are clever enough to manage EBS for you"""
-        instances = super().get_desired_instances()
+
+        if self.get_desired_state() == "start":
+            instances = self.config_dict.get("instances") or self.get_min_instances()
+        else:
+            instances = 0
+            log.debug("Instance is set to stop. Returning '0' instances")
```

Comment:
Shouldn't the state be explicitly compared with "stop" instead of assuming that it's "stop" since it isn't "start"?